lib/routes/route.ex

```diff
@@ -50,4 +50,12 @@ defmodule Routes.Route do
     do: short_name
 
   def name(%__MODULE__{long_name: long_name}), do: long_name
+
+  @spec name_by_id([t()], String.t()) :: String.t()
+  def name_by_id(routes, route_id) do
```

Comment:
suggestion: It feels to me like this is conflating two operations that are working on two different types of data: a collection of routes and then a single route. I.e., having a function in `Route` that accepts a list of routes seems discordant. I think it might feel cleaner if you were to do something like add a `Routes.get(routes, route_id)` function to `Routes` that you can use to retrieve the single route for the given ID given a list of all routes. Then I think the existing `Route.name` function could work on the return value of that.